nucypher/characters/lawful.py

```diff
@@ -1185,6 +1189,22 @@ def __prune_arrangements(self) -> None:
             if result > 0:
                 self.log.debug(f"Pruned {result} policy arrangements.")
 
+        try:
+            with self.datastore.query_by(DatastoreTreasureMap,
+                                         filter_field='expiration',
+                                         filter_func=lambda expiration: expiration <= now,
+                                         writeable=True) as expired_treasure_maps:
+                for treasure_map in expired_treasure_maps:
+                    treasure_map.delete()
+                result = len(expired_treasure_maps)
+        except RecordNotFound:
+            self.log.debug("No expired treasure maps found.")
+        except DatastoreTransactionError:
+            self.log.warn(f"Failed to prune expired treasure maps; DB session rolled back.")
+        else:
+            if result > 0:
+                self.log.debug(f"Pruned {result} treasure maps.")
+
```

Comment:
:+1:  Good keeping an eye out for unhandled crash behavior here.   Even if this method or service fails, the ursula will need to continue running. Once this is out on testnet for a few periods, we'll get a better sense of how stable it is!